src/common/opencl.c

```diff
@@ -735,6 +735,25 @@ static gboolean _opencl_device_init(dt_opencl_t *cl,
     goto end;
   }
 
+  if(strlen(deviceversion) < 9)
+  {
+     dt_print_nts(DT_DEBUG_OPENCL,
+                 "   *** no proper device version ***\n");
+    res = TRUE;
+    goto end;
+  }
+  else
+  {
+    if((strncmp(deviceversion+7, "1.0", 3) == 0)
```

Comment:
Are we safe trying to skip OpenCL prefix here (+7)? It seems very fragile and will break as soon as some new format is used. Aslo isn't the deviceversion here "OpenCL 3.0 CUDA" and not the OpenCL C API version, clinfo gives me:
```

  Device Version                                  OpenCL 3.0 CUDA
  Device OpenCL C Version                         OpenCL C 1.2 
```

Reply:
We are safe as the OpenCL docs define the first 7 chars to be constant. Also safe for introducing new version as we only check for <1.2

As this is definitely a corner case I have no problem with not merging at all.

Reply:
Thanks for clarifying the offset of 7 chars. My last question was not answered, is `deviceversion` the **"Device Version**" or the "**Device OpenCL C Version**"?

Reply:
The spec defines the returned string to be "OpenCL" followed by one of the version id's "1.0, 1.1, 1.2, 2.0..." followed by space followed by some vendor specific stuff.

It's neither the api version nor some device-hardware version but reflects the supported opencl version.

So if we would bump the api to 2.0 we would have to exclude "1.2" too.